eea/facetednavigation/subtypes/subtyper.py

```diff
@@ -22,16 +22,23 @@
     FacetedEnabledEvent,
 )
 from eea.facetednavigation import EEAMessageFactory as _
+from zope.publisher.interfaces import IPublishTraverse
 
 
+@implementer(IPublishTraverse)
 @implementer(IFacetedSubtyper)
 class FacetedPublicSubtyper(BrowserView):
     """ Public support for subtyping objects
         view for non IPossibleFacetedNavigable objects
     """
-    def __init__(self, context, request):
-        self.context = context
-        self.request = request
+    def publishTraverse(self, request, name):
```

Comment:
Since https://github.com/zopefoundation/Zope/issues/397 is fixed in Zope 4.1 we could go back to the original implementation with `allowed_interface=".interfaces.IFacetedSubtyper"`.

Reply:
:+1: